partition_snapshot_reader.hh

```diff
@@ -325,6 +334,18 @@ private:
         }
     }
 
+    static position_in_partition_view for_range_end_in_half_reversed(const query::clustering_range& r) {
+        if constexpr (Reversing) {
+            // TODO: use reversed<clustering_key_prefix>. If for_range_end(reversed<clustering_key_prefix>(r))
+            // was returned, this constitues a return of a reference to an automatic variable.
+            auto bv = bound_view::from_range_start(r);
+            bound_view bvv = {bv.prefix(), reverse_kind(bv.kind())};
+            return {position_in_partition_view::range_tag_t(), bvv};
```

Comment:
This should be simpler:

```
return position_in_partition_view::for_range_start(r).reversed();
```

See the patch on the mailing list:  `[PATCH] position_in_partition: Introduce reversed() transformation`

Reply:
If you switch to native order like suggested in another comment, then there's no need for this wrapper.

Reply:
If I switch to native format from the beginning, I will still need to do the reversing of ck_ranges sometimes, as functions such as `lsa_partition_reader::do_refresh_state` call `lower_bound`:
```
return make_iterator(v.partition().lower_bound(*_snapshot_schema, ck_range));
```
and this `lower_bound` expects that the `ck_range` is in format native to the `_snapshot_schema`. Given that usually I expect slice to be half-reversed (i.e. the order of ck_ranges is reversed, the ranges themselves aren't), I don't see how having this wrapper at all is unreasonable.
This of course doesn't mean that I can't use your patch, which I will.